src/rpc/rpc.h

```diff
@@ -0,0 +1,130 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file rpc/rpc.h
+ * \brief Common headers for remote process call (RPC).
+ */
+#ifndef DGL_RPC_RPC_H_
+#define DGL_RPC_RPC_H_
+
+#include <dgl/runtime/object.h>
+#include <dgl/runtime/ndarray.h>
+#include <dmlc/thread_local.h>
+#include <cstdint>
+#include <memory>
+#include "./network/communicator.h"
+#include "./server_state.h"
+
+namespace dgl {
+namespace rpc {
+
+/*! \brief Context information for RPC communication */
+struct RPCContext {
+  /*!
+   * \brief Rank of this process.
+   *
+   * If the process is a client, this is equal to client ID. Otherwise, the process
+   * is a server and this is equal to server ID.
+   */
+  int32_t rank = -1;
+
+  /*!
+   * \brief Message sequence number.
+   */
+  int64_t msg_seq = 0;
+
+  /*!
+   * \brief Sender communicator.
+   */
+  std::shared_ptr<network::Sender> sender;
```

Comment:
how does it work when it connects with many servers? it seems one process has one RPCContext.

Reply:
`network::Sender` and `network::Receiver` support multiple connections at the same time. @aksnzhy pls corrects me if i'm wrong.

Reply:
Yes. Both sender and receiver can support multiple connections.